### PR TITLE
add more test coverage for unit tests

### PR DESCRIPTION
**Pull request by glorialimicrosoft**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTests.cs [hunk 1]
```diff
@@ -196,6 +217,387 @@ await foreach (Page<ChatMessageReadReceipt> page in allreadReceipts.AsPages(page
...
+            var chatClient = new ChatClient(uri, communicationTokenCredential, chatClientOptions);
+            ChatThreadClient chatThreadClient = chatClient.GetChatThreadClient(threadId);
+            Response updateMessageResponse = await chatThreadClient.UpdateMessageAsync(messageId, content);
+
+            //assert
+            Assert.AreEqual(204, updateMessageResponse.Status);
+        }
+
+        [Test]
+        public async Task GetMessagShouldSucceed()
```
Comment (juancamilor) [inline sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTests.cs:366]: GetMessage

**Review by juancamilor — APPROVED** [review]
Thanks for adding coverage @glorialimicrosoft 
We should probably add some tests for each of the message content types as well (topicUpdated, participantAdded etc..)
And also we can now remove the <Nullable>enable</Nullable> from the project, and update the required code.

File: sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTests.cs [hunk 2]
```diff
@@ -196,6 +217,387 @@ await foreach (Page<ChatMessageReadReceipt> page in allreadReceipts.AsPages(page
...
+        {
+            //arrange
+            var messageId = "1";
+            var content = "Update Message Test";
+            var threadId = "19:e5e7a3fa5f314a01b2d12c6c7b37f433@thread.v2";
+            var uri = new Uri("https://localHostTest");
+            var communicationTokenCredential = new CommunicationTokenCredential(ChatRecordedTestSanitizer.SanitizedUnsignedUserTokenValue);
+            var mockResponse = new MockResponse(204);
+
+            var chatClientOptions = new ChatClientOptions
```
[the remark this answers is not in the document]
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTests.cs:351]: You can have a method and do something like `CreateMockChatClientWithResponse(204)`.

[the remark this answers is not in the document]
Comment (glorialimicrosoft) [reply sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTests.cs]: added

File: sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTests.cs (hunk 1 above)
Comment (glorialimicrosoft) [reply sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTests.cs]: typo fixed

**Comment by juancamilor** [comment]
Do you want to remove the <Nullable>enable</Nullable> from the project in a follow up PR @glorialimicrosoft ?
Either way works. Thanks for adding coverage !

**Review by RezaJooyandeh — APPROVED** [review]

**Comment by glorialimicrosoft** [comment]
> Thanks for adding coverage @glorialimicrosoft
> We should probably add some tests for each of the message content types as well (topicUpdated, participantAdded etc..)
> And also we can now remove the enable from the project, and update the required code.

add some tests for each of the message content types

**Pull request merged**